scrapy/http/request/json_request.py

```diff
@@ -0,0 +1,28 @@
+"""
+This module implements the JSONRequest class which is a more convenient class
+(than Request) to generate JSON Requests.
+
+See documentation in docs/topics/request-response.rst
+"""
+
+import json
+
+from scrapy.http.request import Request
+
+
+class JSONRequest(Request):
+    def __init__(self, *args, **kwargs):
+        if 'method' not in kwargs:
+            kwargs['method'] = 'POST'
+
+        data = kwargs.pop('data', {})
```

Comment:
I had a look into ``XmlRpcRequest ``, which seems quite close in purpose:
https://github.com/scrapy/scrapy/blob/886513c3751b92e42dcc8cb180d4c15a5a11ccaf/scrapy/http/request/rpc.py#L20

It's approach makes more sense to me:
- it allows passing raw ``body``
- it allows passing ``params`` (I'd rather also have ``params`` here for consistency) if no ``body`` is given
- if nothing is passed, it uses usual scrapy default for ``body``, and not an empty dict which seems arbitrary to me

Reply:
Regarding the naming: maybe ``data`` is still better, because e.g. requests lib uses ``data`` for this purpose, while ``params`` means a different thing.

Reply:
Thanks @lopuhin for the quick review. Now,
body is allowed to be passed
data is ignored if body is passed closely resembling the XmlRpcRequest